src/afrolabs/components/kafka.clj

```diff
@@ -742,18 +742,21 @@
           [_ _]
         (csp/close! caught-up-ch))
 
-      IConsumerPostInitHook
-      (post-init-hook
-          [_ consumer]
-        ;; We save the end offsets at the start, so we can compare later if we've caught up
-        ;; If this times out, the exception will be bubbled up.
-        (reset! end-offsets-at-start
-                (consumer-end-offsets consumer :timeout-value :throw)))
-
       IPostConsumeHook
       (post-consume-hook
           [_ consumer _consumed-records]
+        ;; we have to store the endOffsets, after the first consume (ie after partition assignment)
+        ;; if we have a value in this atom, just keep it
+        ;; otherwise query the end offsets and store it
+        (swap! end-offsets-at-start
+               (fn [old-value]
+                 (if old-value
+                   old-value
+                   (consumer-end-offsets consumer :timeout-value :throw))))
```

Comment:
Very late in the game to discover this bug. For some ktables (those configure with `:caught-up-once true`, we need to wait on the ktable until it has loaded at least as many records as it had when it starts up. When you read the end offsets on assigned partitions, before the first .poll, you get nothing back. Therefore this logic never did what it was intended to do. 

We change the implementation here to read the endoffsets, ONCE, after the first .poll / consume.